src/lib/common.js

```diff
@@ -0,0 +1,23 @@
+/**
+ * Polls a function until its return value satisfies a condition
+ * @param       {function} fn A function to poll
+ * @param       {int} interval Poll interval in milliseconds
+ * @param       {function} isDone A function that returns true if the polling should stop
+ * @resolves		{Promise} A promise which resolves when the polling is done
+ */
+export async function pollUntil( fn, interval, isDone ) {
+	return new Promise( async ( resolve, reject ) => {
```

Comment:
You'll want `return await new Promise` here. See [ref](https://github.com/goldbergyoni/nodebestpractices/blob/master/sections/errorhandling/returningpromises.md), pretty surprising but that's how it is.

Reply:
See also: https://eslint.org/docs/latest/rules/no-async-promise-executor

I think if you update the branch, ESLint will complain.

Reply:
> Would it be cleaner to just not explicitly return a new Promise?

That's a very good idea. Wonder how I missed that :smile: